src/services/group.service.ts

```diff
@@ -0,0 +1,57 @@
+import { ModelCtor, Sequelize } from 'sequelize';
+import { Inject, Service } from 'typedi';
+
+import { Config } from '../config';
+import { CreateActionGroup, UpdateActionGroup } from '../interfaces';
+import { IGroupInstance } from '../models';
+
+@Service()
+export class GroupService {
+    constructor(
+        @Inject(Config.injectionToken.groupModel) private groupModel: ModelCtor<IGroupInstance>,
+        @Inject(Config.injectionToken.sequelize) private sequelize: Sequelize,
+    ) {}
+
+    async getGroupById(id: string) {
+        return await this.sequelize.transaction(() => {
+            return this.groupModel.findOne({ where: { id } });
+        });
+    }
+
+    async getAllGroups() {
+        return await this.sequelize.transaction(() => {
+            return this.groupModel.findAll();
+        });
+    }
+
+    async creatGroup(createActionGroup: CreateActionGroup) {
+        return await this.sequelize.transaction(() => {
+            return this.groupModel.create({ ...createActionGroup });
+        });
+    }
+
+    async updateGroup(groupId: string, updateActionGroup: UpdateActionGroup) {
+        return await this.sequelize.transaction(() => {
+            return this.groupModel.update(
+                {
+                    ...updateActionGroup,
+                },
+                {
+                    where: {
+                        id: groupId,
+                    },
+                },
+            );
+        });
+    }
+
+    async deleteGroup(groupId: string) {
+        return await this.sequelize.transaction(() => {
```

Comment:
It doesn't make sense to me to execute single operation within a transaction. What's the point?

Reply:
@KonanMentor yes, I agree with you, I have removed it in places where only one query is used